src/coreclr/jit/lsrabuild.cpp

```diff
@@ -3310,6 +3314,40 @@ int LinearScan::BuildBinaryUses(GenTreeOp* node, regMaskTP candidates)
     return srcCount;
 }
 
+//------------------------------------------------------------------------
+// BuildCastUses: Build uses for a cast's source, preferencing it as appropriate.
+//
+// Arguments:
+//    cast       - The cast node to build uses for
+//    candidates - The candidate registers for the uses
+//
+// Return Value:
+//    The number of actual register operands.
+//
+int LinearScan::BuildCastUses(GenTreeCast* cast, regMaskTP candidates)
+{
+    GenTree* src = cast->CastOp();
+
+    // Casts can have contained memory operands.
+    if (src->isContained())
+    {
+        return BuildOperandUses(src, candidates);
+    }
+
+    RefPosition* srcUse = BuildUse(src, candidates);
+
+#ifdef TARGET_64BIT
+    // A long -> int cast is a copy - the code generator will elide
+    // it if the source and destination registers are the same.
```

Comment:
Can you remind me why this is true, and why we don't need to zero or sign-extend the top 32 bits?

Is it because on non-LA64 we always generate 32-bit register references for the subsequent uses?

Reply:
> Is it because on non-LA64 we always generate 32-bit register references for the subsequent uses?

Yes, and if the value ends up in memory (spilled), we will only load 32 bits.

The general logic is that for a `TYP_INT` value in IR, the upper bits, if they exist, are undefined.